.github/workflows/main.yml

```diff
@@ -1,17 +1,29 @@
 name: C++ with CMake CI
 
-on: [push, pull_request]
+on:
+ 
+  push:
+    branches:
+      - homework1
+  pull_request:
+    branches:
+      - homework1
 
 jobs:
-  checker:
+  shm:
     runs-on: ubuntu-latest
-    steps:
-    - name: Check out code
-      uses: actions/checkout@master
-    - name: Run check.sh
-      working-directory: shm
-      run: |
-        chmod +x check.sh
-        ./check.sh CMakeLists.txt
+    steps:     
+     - name: checkout
+       uses: actions/checkout@master
+     - name : mkdir shm/build
+       run: mkdir shm/build
+     - name: Compile
+       working-directory: shm/build
+       run: |
+        cmake ..
+        make
+     - name: Run tests
+       working-directory: shm/build
+       run: ctest -V
+
 
-# In case of conflict do not remove above jobs. Just add your jobs below and remove this comment :)
```

Comment:
Hello, have you read this comment? :D

Reply:
yes, i've messed up with main.yml, and really I do not remember how I ended up with this merge... I can give my word that it was not on purpose. :)